MCR-2339 login ArrayIndexOutOfBoundException if you enter a Username with @ as last char

OPNA2608 opened the pull request:
[Link to jira](https://mycore.atlassian.net/browse/MCR-2339).

OPNA2608 committed:
MCR-2339 login ArrayIndexOutOfBoundException if you enter a Username with @ as last char

yagee-de reviewed (changes requested):
I do not think we should allow an empty realmId here
if `parts.length != 2` throw IllegalArgumentException here

sebhofmann commented:
> I do not think we should allow an empty realmId here
> if `parts.length != 2` throw IllegalArgumentException here

Also second "@" is possible.

OPNA2608 commented:
> if `parts.length != 2` throw IllegalArgumentException here

I don't think replacing one MyCoRe kernel exception screen with a slightly different one is the best UX solution here.

![Bildschirmfoto von 2021-02-18 12-17-25](https://user-images.githubusercontent.com/23431373/108349796-b732b280-71e3-11eb-9a0c-2e749cbcec18.png)

---

What if I do this instead:
```java
if (parts.length == 2) {
    return getUser(parts[0], parts[1]);
} else {
    return null;
}
```

Since `checkPassword` already tests for `getUser(userName) == null, that should work without pushing an empty realmId through the realm checking code.

OPNA2608 committed:
MCR-2339 login ArrayIndexOutOfBoundException if you enter a Username with @ as last char

Return null (invalid) if username contains @ but splits in more/less than 2 parts
